sweetspi-processor/src/main/kotlin/analyze.kt

```diff
@@ -42,9 +42,26 @@ fun analyze(logger: KSPLogger, resolver: Resolver): SweetContext? {
             // TODO: check visibility
             when (it) {
                 is KSClassDeclaration    -> {
-                    if (it.classKind != ClassKind.OBJECT) {
+                    if (it.classKind == ClassKind.OBJECT) {
+                        isValid = true
+                    } else if (it.classKind == ClassKind.CLASS) {
+                        val ctors = it.getConstructors().toList()
+                        // if there are no constructors, a default 0-arg constructor will be generated by Kotlin compiler
+                        if (ctors.isEmpty() || ctors.any { it.parameters.all { it.hasDefault } }) {
+                            isValid = true
+                        } else {
+                            isValid = false
+                            logger.error(
```

Comment:
It would be nice to have this, and the other error on line 61 same. saying that `@ServiceProvider target '${it.simpleName.asString()}' should be an 'object' or 'class' with a constructor that has no parameters or a constructor with all parameters having default value`. (similar/same to what annotation KDoc says)
`defaultable` is not an understandable term, IMO.

Ideally, this message should not be duplicated in multiple `logger.error` calls.